File: Week 4/A/binary.py
L = 1 
R = N
ans = 0 
while(L<=R): # while there are still subarrays
    mid = (L+R)//2 # rounded down 
    if(arr[mid]<K):
        ans = arr[mid]
        L = mid + 1 
    else:
        R = mid - 1

File: 2020-2021/Week 4/A/binary.cpp
int L = 1, R = N, ans = 0;
while(L<=R) {
    int mid = (L+R)/2; 
    if(arr[mid] < K) {
        ans = arr[mid]; 
        L = mid+1; 
    }
    else {
        R = mid-1; 
    }
}

